TinCan/ActivityDefinition.cs

```diff
@@ -85,6 +146,70 @@ public ActivityDefinition(JObject jobj)
             {
                 result.Add("extensions", extensions.ToJObject(version));
             }
+            if (interactionType != null)
+            {
+                result.Add("interactionType", interactionType.ToString());
+            }
+            if (correctResponsesPattern != null && correctResponsesPattern.Count > 0)
```

Comment:
We just hit this on the spec, but we need to leave it open to allow for an empty array in 'correctResponsesPattern', see https://github.com/adlnet/xAPI-Spec/issues/698